src/common/sfserr/saunafs_error_codes.cc

```diff
@@ -19,8 +19,6 @@
    along with SaunaFS  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "common/platform.h"
-
```

Comment:
The common/platform.h contains parts related to the error codes, like this:
```cpp
/* Workaround for Debian/kFreeBSD which does not define ENODATA */
#if defined(__FreeBSD_kernel__) || defined(__FreeBSD__)
#ifndef ENODATA
#define ENODATA ENOATTR
#endif
#endif
```
Please check the implications of not including the header.

Reply:
I'll be moderately surprised if it actually builds on FreeBSD even without this change, but I'll check it out.

Reply:
I think the build is generally broken on FreeBSD, I couldn't get main to compile at all, so this is the least of our worries. As for the error code itself which is indeed undefined on FreeBSD, I think the easiest solution is just to copy this portion to the library header, so only the libraries that actually need this will have it defined.

`common/platform.h` seems to be full of hacks though, It needs documenting on what is actually needed and why, and removing things which are not.

Reply:
e.g to_string is implemented there for platforms that are missing it and was committed 10 years ago, but are there any platforms still missing it and if so, what platforms?

Reply:
The selected code was just an example. The main idea is that the common/platform.h file has a purpose, and that's why it is included first in every file, it also includes config.h. If you are planning to change it, then this is not the commit for it.

Reply:
Ok, I will revert it, but I was hoping to discuss it with these changes when we are splitting the common library up. I don't like the fact that platform.h is modifying the std library, including headers and doing a bunch of hacks which may not be needed by every library. At the very least, new libraries should not include it unless you have a good reason to.

Reply:
Looks like we need to review the platform.h file and update it.